### PR TITLE
Impact study (broken)

### PR DESCRIPTION
**Pull request by jrgerber**
@mulark here's a quick impact study between dmidecode-rs and smbios-lib 0.9.0 for 3.5.0.  Code currently broken, just pushing this as an overview to look at.

### CODE REVIEW

File: src/default_out.rs [hunk 1]
```diff
@@ -1141,15 +1145,15 @@ pub fn dump_undefined_struct(
                 );
             }
             for installable_language in data.installable_langauges() {
```
Comment (mulark) [inline src/default_out.rs:1147]: Shouldn't SMBiosStringSet should iterate over elements of SMBiosString? Instead of `Result<String, FromUtf8Error>`

**Comment by jrgerber** [comment]
It's a good question.  When you iterate the set you know you should only
get strings or failed UTF-8 parse errors but not errors of out-of-bounds or
bad index like you would if given an index.  I'm going to sleep on this one
tonight.  After making this an SMBiosStringSet with SMBiosStrings it feels
odd having to answer the match arms.

File: src/default_out.rs (hunk 1 above)
Comment (jrgerber) [reply src/default_out.rs:1151]: OK, have a look now.  This should build without errors.  With the new SMBiosString being returned, I replaced these spots with a .to_utf8_lossy().unwrap_or_default().  I also added code for handling the new CacheSize...

Comment (mulark) [reply src/default_out.rs:1151]: Oh, the I already implemented cache size fixes in https://github.com/jrgerber/dmidecode-rs/pull/54. But I think this iterator change is the right one.

**Pull request closed without merging**